src/query/execute_schema.c

```diff
@@ -10016,6 +10020,366 @@ do_alter_change_tbl_comment (PARSER_CONTEXT * const parser, PT_NODE * const alte
+	     if (saved_constr->func_index_info)
+	       {
+		 error = do_recreate_func_index_constr (parser, saved_constr, NULL, alter, NULL, NULL);
+		 if (error != NO_ERROR)
+		   {
+		     goto exit;
+		   }
+	       }
+	     if (saved_constr->filter_predicate)
+	       {
+		 error = do_recreate_filter_index_constr (parser, saved_constr->filter_predicate, alter, NULL, NULL);
+		 if (error != NO_ERROR)
+		   {
+		     goto exit;
+		   }
+	       }
+
+	     if (!is_srv_update_needed)
+	       {
+		 const char *att_names[2];
+		 PT_NODE *att_old_name = alter->info.alter.alter_clause.attr_mthd.attr_old_name;
+
+		 if (att_old_name != NULL)
+		   {
+		     assert (att_old_name->node_type == PT_NAME);
+		     att_names[0] = att_old_name->info.name.original;
+		     att_names[1] = NULL;
+
+		     assert (alter->info.alter.alter_clause.attr_mthd.attr_old_name->node_type == PT_NAME);
+		     error =
+		       sm_drop_constraint (class_mop, saved_constr->constraint_type, saved_constr->name, att_names,
+					   false, false);
+
+		     if (error != NO_ERROR)
+		       {
+			 goto exit;
+		       }
+
+		     error = sm_add_constraint (class_mop, saved_constr->constraint_type, saved_constr->name,
+						(const char **) saved_constr->att_names, saved_constr->asc_desc,
+						saved_constr->prefix_length, false, saved_constr->filter_predicate,
+						saved_constr->func_index_info, saved_constr->comment,
+						saved_constr->index_status);
+		     if (error != NO_ERROR)
+		       {
+			 goto exit;
+		       }
+		   }
+	       }
+	   }
+       }
+    }
+
+  if (is_srv_update_needed || is_att_prop_set (attr_chg_prop.p[P_TYPE], ATT_CHG_TYPE_PREC_INCR))
+    {
+      error = do_drop_att_constraints (class_mop, attr_chg_prop.constr_info);
+      if (error != NO_ERROR)
+       {
+	 goto exit;
+       }
+
+      /* perform UPDATE or each row */
+      if (is_srv_update_needed)
+       {
+	 assert (att_id >= 0);
+	 assert (!OID_ISNULL (&class_oid));
+
+	 if (has_partitions)
+	   {
+	     assert (user_count > 0);
+	     assert (usr_oid_array != NULL);
+
+	     for (i = 0; i < user_count; i++)
+	       {
+		 error = do_run_upgrade_instances_domain (parser, &(usr_oid_array[i]), att_id);
+		 if (error != NO_ERROR)
+		   {
+		     goto exit;
+		   }
+	       }
+	   }
+	 else
+	   {
+	     error = do_run_upgrade_instances_domain (parser, &class_oid, att_id);
+	     if (error != NO_ERROR)
+	       {
+		 goto exit;
+	       }
+	   }
+       }
+
+      error = sort_constr_info_list (&(attr_chg_prop.constr_info));
+      if (error != NO_ERROR)
+       {
+	 er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_UNEXPECTED, 1, "Sorting constraints failed.");
+	 goto exit;
+       }
+
+      error = do_recreate_att_constraints (class_mop, attr_chg_prop.constr_info);
+      if (error != NO_ERROR)
+       {
+	 goto exit;
+       }
+    }
+  else
+    {
+      assert (change_mode == SM_ATTR_CHG_ONLY_SCHEMA);
+    }
+
+  /* create any new constraints: */
+  if (attr_chg_prop.new_constr_info != NULL)
```

Comment:
**new_constr_info** is set at adding a new constraint, so I think that this conditional codes are not necessesary.

Reply:
Thank you. The code has been removed at the final commit.